### ISSUE TITLE
Acl inheritance wildcard

### ISSUE THREAD

**Issue by KuLi** [phalcon/cphalcon#12004]
Hi, i inherit from role and check for a non existing resource i get deny on wildcard check.

In my case a guest can access payment with resource paypal. A user can access all from payment and also an admin and superadmin. The user can access the not existing resource but the admin can't. 

Here my testcase
```php
<?php

$oAcl = new \Phalcon\Acl\Adapter\Memory();

$oAcl->setDefaultAction(\Phalcon\Acl::DENY);

$oRoleGuest = new \Phalcon\Acl\Role("guest");
$oRoleUser = new \Phalcon\Acl\Role("user");
$oRoleAdmin = new \Phalcon\Acl\Role("admin");
$oRoleSuperAdmin = new \Phalcon\Acl\Role("superadmin");

$oAcl->addRole($oRoleGuest);
$oAcl->addRole($oRoleUser, $oRoleGuest);
$oAcl->addRole($oRoleAdmin, $oRoleUser);
$oAcl->addRole($oRoleSuperAdmin, $oRoleAdmin);

$oAcl->addResource("payment", array("paypal", "facebook",));

$oAcl->allow($oRoleGuest->getName(), "payment", "paypal");
$oAcl->allow($oRoleGuest->getName(), "payment", "facebook");

$oAcl->allow($oRoleUser->getName(), "payment", "*");

var_dump(
    $oAcl->isAllowed($oRoleUser->getName(), "payment", "notSet"),
    $oAcl->isAllowed($oRoleUser->getName(), "payment", "*"),
    $oAcl->isAllowed($oRoleAdmin->getName(), "payment", "notSet"),
    $oAcl->isAllowed($oRoleAdmin->getName(), "payment", "*")
);
```

My output
```text
true

true

false

true
```

Version 2.1.0 RC 1

**Comment by Jurigag** [comment]
Obviously it's not checked - https://github.com/phalcon/cphalcon/blob/master/phalcon/acl/adapter/memory.zep#L542

**Comment by Jurigag** [comment]
Checked on 2.0.x and happening as well.

**Comment by Jurigag** [comment]
I think that we can remove this if:

https://github.com/phalcon/cphalcon/blob/2.0.x/phalcon/acl/adapter/memory.zep#L382 AND
https://github.com/phalcon/cphalcon/blob/2.0.x/phalcon/acl/adapter/memory.zep#L406 this one

And change https://github.com/phalcon/cphalcon/blob/2.0.x/phalcon/acl/adapter/memory.zep#L619 to:

```
return (this->_defaultAccess == ACL::Allow);
```

**Comment by Jurigag** [comment]
Well created PR, but it's changing somehow internally how acl was working. Not sure if it will affect any app.

**Comment by sergeyklay** [comment]
Fixed in the `2.1.x` branch.
